kernel/src/process.rs

```diff
@@ -830,10 +892,14 @@ pub struct ProcessAddresses {
     /// nonvolatile memory. This is after the TBF header and any other memory
     /// the kernel has reserved for its own use.
     pub flash_non_protected_start: usize,
+    /// The address immediately after the end of part of the process
+    /// binary that is covered by integrity; the integrity region is
+    /// [flash_start - flash_integrity_end). Footers are stored in
+    /// the flash after flash_integrity_end.
+    pub flash_integrity_end: *const u8,
```

Comment:
Should this be `binary_end_offset` or `flash_binary_end` for consistency? Using "integrity" makes it unclear if this is just the end of app binary or if this is where integrity protection actually ended (and if there was no integrity check than this would equal `flash_start`, or something).

Reply:
The goal is to store this particular value -- the address after the region covered by integrity. It's not where integrity protection ends (which would imply an integrity check), but rather the end of the region which integrity protects. It's not `binary_end_offset` because that's a particular header field; I don't want to bind this value to the representation used for `binary_end_offset`. E.g., if we decided `binary_end_offset` should be an offset from the end of the Header portion, this should clearly not be that. I also want to keep it distinct from `flash_binary_end`, in case we decide that there needs to be some kind of a fixed footer after the Userspace Binary.